src/components/IKImage/IKImage.js

```diff
@@ -33,31 +33,83 @@ class IKImage extends ImageKitComponent {
     return state;
   }
 
-  lqipload(quality) {
+  lqipload(quality, blur, threshold) {
+    const finalQuality = parseInt((quality || threshold), 10) || 20;
+    const finalBlur = parseInt(blur, 10) || 6;
     let url = this.state.url;
     let lqip = "";
     if (this.props.path !== undefined) {
       let newUrl = url.split("tr:");
       if (newUrl[0] === url) {
         let newUrl = url.split("/");
-        newUrl = `${newUrl[0]}//${newUrl[2]}/${newUrl[3]}/tr:q-${quality}/${newUrl[4]}`;
+        newUrl = `${newUrl[0]}//${newUrl[2]}/${newUrl[3]}/tr:q-${finalQuality},bl-${finalBlur}/${newUrl[4]}`;
         lqip = `${newUrl}`;
       } else {
-        newUrl = `${newUrl[0]}tr:q-${quality},${newUrl[1]}`;
+        newUrl = `${newUrl[0]}tr:q-${finalQuality},bl-${finalBlur},${newUrl[1]}`;
         lqip = `${newUrl}`;
       }
     } else {
       if(url.includes("tr")){
-        lqip = `${url}&q-${quality}`;
+        lqip = `${url}&q-${finalQuality},bl-${finalBlur}`;
       }
       else {
-        lqip = `${url}&tr=q-${quality}`;
+        lqip = `${url}&tr=q-${finalQuality},bl-${finalBlur}`;
       }
 
     }
     return lqip;
   }
 
+  getFinalImageSrcURL(loading, lqip){
+
+    /*
+    No lazy loading no lqip
+      src=originalImage
+    No lazy loading lqip
+      src=lqip
+      src=originalImage (when loaded)
+    lazy loading and no lqip
+      src=''
+      onIntersect:
+      src=originalImage
+    lazy loading and lqip
+      src=lqip
+      onInterserct:
+      src=originalImage (when loaded)
+   */
+
+    const {
```

Comment:
Does `this.imageRef.current` have `originalSrcLoaded`?

In Vue.js SDK, we are setting `originalSrcLoaded` manually once the original image download has complete in the background. The current implementation won't replicate the same behavior as Vue.js.

Reply:
yeah i had a question about this specifically. should i handle this in the state with an onLoad handler? im assuming that the observer handles intersection but want to know how to handle these cases. @imagekitio

Reply:
This is how it works in Vue.js

We observe the current element `this.observer.observe(this.$el);` if `loading="lazy"` is passed. Checkout `mounted` function definition in file - https://github.com/imagekit-developer/imagekit-vuejs/blob/master/src/components/IKImage.vue

Now once we have `isIntersecting=true`, we trigger original image load if `lqip=true`
```
if (image && image.isIntersecting) {
   this.intersected = true;
   if(this.lqip) this.triggerOriginalImageLoad();
   this.observer.disconnect();
}
```

`triggerOriginalImageLoad` function loads the original image and set `originalSrcLoaded` to true using `onload` event on `img` tag.

The same has to be replicated in React SDK.

Reply:
I have replicated the logic. Have kept an onLoad handler to set the state once the image has loaded and in the observer have added isIntersecting to be true as was done in the earlier implementation but have set a state variable called isIntersecting to be set once the image actually loads. Could you look into the approach and see if there is any serious drawback to this @imagekitio

Reply:
@imagekitio any updates on this?

Reply:
We are relying on the `onLoad` event but we don't know if it will be triggered after low quality image is loaded or the original src. We can make this logic explicit as in Vue.js by triggering the `triggerOriginalImageLoad` function appropriately.